glean-core/src/lib.rs

```diff
@@ -101,6 +102,10 @@ pub fn setup_glean(glean: Glean) -> Result<()> {
             log::warn!(
                 "Global Glean object is initialized already. This probably happened concurrently."
             )
+        } else {
+            // Can only kick off the "metrics" ping scheduler after we have a global Glean.
+            let glean = &GLEAN.get().unwrap().lock().unwrap();
+            scheduler::schedule(&glean);
```

Comment:
I would probably swap around the `if` condition here:

```rust
if GLEAN.set(Mutex::new(glean)).is_ok() {
	// Can only kick off the "metrics" ping scheduler after we have a global Glean.
    let glean = &GLEAN.get().unwrap().lock().unwrap();
    scheduler::schedule(&glean);
} else {
    // ...
}
```

That is: handle the error case last